src/wp-includes/class-wp-scripts.php

```diff
@@ -281,6 +300,26 @@ public function do_item( $handle, $group = false ) {
 			$intended_strategy = '';
 		}
 
+		/*
+		 * Move this script to the footer if:
+		 * 1. The script is in the header group.
+		 * 2. The current output is the header.
+		 * 3. The intended strategy is delayed.
+		 * 4. The actual strategy is not delayed.
+		 * 5. All dependent scripts are in the footer.
+		 */
+		if (
+			0 === $group &&
+			0 === $this->groups[ $handle ] &&
+			$intended_strategy &&
+			! $this->is_delayed_strategy( $strategy ) &&
+			$this->are_all_dependents_in_footer( $handle )
+		) {
+			$this->in_footer[] = $handle;
+			$this->set_group( $handle, 1 );
```

Comment:
Commenting out this line results in the tests still passing:

```suggestion
```

Reply:
yes I know because we aren't testing for the group.

Interestingly when I tried writing a test for group and looked at `wp_script()->groups` it didn't look like it had changed (so the tests didn't work). I'm fine removing this line or trying to expand the tests to cover it.